plugin/src/main/kotlin/com/ritense/plugin/domain/PluginConfiguration.kt

```diff
@@ -51,4 +55,20 @@ class PluginConfiguration(
             Mapper.INSTANCE.get().treeToValue(properties, T::class.java)
         }
     }
+
+    fun updateProperties(propertiesForUpdate: JsonNode) {
+        if (properties is ObjectNode) {
```

Comment:
If it's not an `ObjectNode`, then it can't be updated. I would expect this method to throw an exception in that case.

Reply:
Since we need the properties to always be an objectnode, i changed the type to reflect that